### PR TITLE
fix(ui5-upload-collection): item-delete event and delete mode docs are now updated

### PR DESCRIPTION
**Pull request by dimovpetar**
Fixes: https://github.com/SAP/ui5-webcomponents/issues/7085

### CODE REVIEW

File: packages/fiori/src/UploadCollection.ts [hunk 1]
```diff
@@ -130,8 +124,10 @@ class UploadCollection extends UI5Element {
 	 * <ul>
 	 * <li><code>None</code></li>
 	 * <li><code>SingleSelect</code></li>
+	 * <li><code>SingleSelectBegin</code></li>
+	 * <li><code>SingleSelectEnd</code></li>
 	 * <li><code>MultiSelect</code></li>
-	 * <li><code>Delete</code></li>
+	 * <li><code>Delete</code> - this mode has no effect. The delete button is controlled by the <code>hideDeleteButton</code> property of UploadCollectionItem</li>
```
Comment (ilhan007) [inline packages/fiori/src/UploadCollection.ts:130]: I suggest describing this in the [[TOPIC] 2.0 breaking changes list ](https://github.com/SAP/ui5-webcomponents/issues/4460) to whatever you would ideally see as the API related to the delete btn and delete mode.

Comment (dimovpetar) [reply packages/fiori/src/UploadCollection.ts:130]: This behavior is introduced with https://github.com/SAP/ui5-webcomponents/pull/3404 and described in  **Components API review** . I'm just fixing some things related to it and making the docs more clear.

Comment (ilhan007) [reply packages/fiori/src/UploadCollection.ts:130]: Yes I know, but we can plan to remove the mode if it does not have effect and purpose

<li><code>Delete</code> - this mode has no effect. The delete button is controlled by the <code>hideDeleteButton</code> property of UploadCollectionItem</li>

or remove the property and leave the Delete mode and so on.

Comment (dimovpetar) [reply packages/fiori/src/UploadCollection.ts:130]: Good point. I have added it.

File: packages/fiori/src/UploadCollection.ts [hunk 2]
```diff
@@ -41,10 +41,6 @@ type SelectionChangeEventDetail = {
 	selectedItems: Array<UploadCollectionItem>,
 };
 
-type ItemDeleteEventDetail = {
-	item: UploadCollectionItem,
-};
```
Comment (gmkv) [inline packages/fiori/src/UploadCollection.ts:46]: What is the intent for removing the type annotation from the event?

Comment (dimovpetar) [reply packages/fiori/src/UploadCollection.ts:46]: It was used by the UploadCollectionItem to send private information to the UploadCollection. This code is now deleted. See @ui5-item-delete and @ui5-_uci-delete in UploadCollection.hbs

Comment (dimovpetar) [reply packages/fiori/src/UploadCollection.ts]: As we discussed this was not a useful type, because the public event was not using it. Now I changed that.

**Review by gmkv — APPROVED** [review]

**Pull request merged**